RecoTracker/FinalTrackSelectors/plugins/TrackSelectorByRegion.cc

```diff
@@ -0,0 +1,87 @@
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/Common/interface/OwnVector.h"
+#include "RecoTracker/TkTrackingRegions/interface/TrackingRegion.h"
+
+#include <vector>
+#include <memory>
+
+class TrackSelectorByRegion final : public edm::global::EDProducer<> {
+public:
+  explicit TrackSelectorByRegion(const edm::ParameterSet& conf)
+      : produce_collection(conf.getParameter<bool>("produceTrackCollection")),
+        produce_mask(conf.getParameter<bool>("produceMask")),
+        tracksToken_(consumes<reco::TrackCollection>(conf.getParameter<edm::InputTag>("tracks"))),
+        inputTrkRegionToken_(consumes<edm::OwnVector<TrackingRegion>>(conf.getParameter<edm::InputTag>("regions"))) {
+    if (produce_collection)
+      produces<reco::TrackCollection>();
+    if (produce_mask)
+      produces<std::vector<bool>>();
+  }
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    desc.add<edm::InputTag>("tracks", edm::InputTag("hltPixelTracks"));
+    desc.add<edm::InputTag>("regions", edm::InputTag(""));
+    desc.add<bool>("produceTrackCollection", true);
+    desc.add<bool>("produceMask", true);
+    descriptions.add("trackSelectorByRegion", desc);
+  }
+
+private:
+  using MaskCollection = std::vector<bool>;
+
+  void produce(edm::StreamID, edm::Event& iEvent, const edm::EventSetup&) const override {
+    if (not produce_collection and not produce_mask)
+      return;
+
+    auto regionsHandle = iEvent.getHandle(inputTrkRegionToken_);
+    auto tracksHandle = iEvent.getHandle(tracksToken_);
+
+    const auto& tracks = *tracksHandle;
+    auto mask = std::make_unique<MaskCollection>(tracks.size(), false);  // output mask
+    const auto& regions = *regionsHandle;
+
+    for (auto const& region : regions) {
+      auto const& region_mask = region.checkTracks(tracks);
+      for (size_t i = 0; i < region_mask.size(); ++i) {
+        mask->at(i) = mask->at(i) or region_mask[i];
```

Comment:
> never use at; it checks boundaries and can throw

I agree when the index is clearly within the boundary, like for `region_mask` since the loop is on its size.
In general, I don't mind the extra check, given the number of times I've seen people access data outside the container :-(

Here I'd be fine changing `mask->at[i]` to `(*mask)[i]` only if we add an `assert(mask->size() == regiona_mask.size())` before the loop...